cli/azd/cmd/container.go

```diff
@@ -596,10 +597,33 @@ func registerCommonDependencies(container *ioc.NestedContainer) {
 		container.MustRegisterNamedSingleton(platformName, constructor)
 	}
 
-	container.MustRegisterSingleton(NewWorkflowRunner)
+	container.MustRegisterSingleton(func(s ioc.ServiceLocator) (workflow.AzdCommandRunner, error) {
```

Comment:
This is sort of verbose, but I wasn't sure if there was a better way to request a named import here, so I have to do the indirect via the Service Locator.  I wish I could just instead inject a *cobra.Command and somehow denote the I wanted the one with the name `root-cmd`)

Reply:
Let me know if you have a better idea how to structure this, @wbreza.

Reply:
Named singletons are tough to inject via constructor injection.  This is the same model I have used in some other components.  LGTM.